crates/ide-completion/src/completions/keyword.rs

```diff
@@ -163,4 +163,75 @@ fn main() {
 "#,
         );
     }
+
+    #[test]
+    fn if_completion_in_match_guard() {
+        check_edit(
+            "if",
+            r"
+fn main() {
+    match () {
+        () $0
+    }
+}
+",
+            r"
+fn main() {
+    match () {
+        () if $0
+    }
+}
+",
+        )
+    }
+
+    #[test]
+    fn if_completion_in_match_arm_expr() {
+        check_edit(
+            "if",
+            r"
+fn main() {
+    match () {
+        () => $0
+    }
+}
+",
+            r"
+fn main() {
+    match () {
+        () => if $1 {
+    $0
+}
+    }
```

Comment:
The indentation here looks wrong, but this test passes.
When I use the assist for this example, the indentation appears correct in vscode. Any idea what's up here?

Reply:
Might be VSCode fixing things up, don't pay formatting any mind in assists and completions. We can't do much about that until we ship our own builtin formatter